packages/synapse-react-client/src/components/DirectDownload/FileEntityDirectDownload.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { useGetEntity } from '../../synapse-queries'
+import { FileHandleAssociateType } from '@sage-bionetworks/synapse-types'
+import { isFileEntity } from '../../utils/types/IsType'
+import DirectDownload, { DirectFileDownloadProps } from './DirectDownload'
+import { Skeleton } from '@mui/material'
+
+export type FileEntityDirectDownloadProps = Omit<
+  DirectFileDownloadProps,
+  'associatedObjectId' | 'associatedObjectType' | 'fileHandleId'
+> & {
+  entityId: string
+  entityVersionNumber?: number
+}
+
+export default function FileEntityDirectDownload(
```

Comment:
New component to handle the FileEntity behavior removed from `DirectDownload`. IMO much simpler to do it this way

Reply:
Agreed, nice refactor